app/src/components/courses/courses.tpl.html

```diff
@@ -0,0 +1,14 @@
+<section ng-controller="CoursesController" class="courses courses__container">
+    <app-course ng-repeat="course in getCourses() | limitTo:coursesLimit
+                                                  | filter: getFilterValue()
+                                                  | orderBy: orderByDate"
+                data="course"
+                ng-if="getCourses()"></app-course>
```

Comment:
As I see, you have three ng-if and one ng-repeat that uses getCourses() here. I believe that these calls can be moved to controller variable. $scope.courses = getCourses()

Reply:
OK. I fixed it.